### PR TITLE
Allowing custom text field name for Milvus (#7789)

### PR DESCRIPTION
**Pull request by ytang07**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # 7789

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: llama_index/vector_stores/milvus.py [hunk 1]
```diff
@@ -91,6 +94,7 @@ def __init__(
         similarity_metric: str = "IP",
         consistency_level: str = "Strong",
         overwrite: bool = False,
+        text_key: str = None,
```
Comment (logan-markewich) [inline llama_index/vector_stores/milvus.py:97]: Should be `Optional[str]`

File: llama_index/vector_stores/milvus.py [hunk 2]
```diff
@@ -270,9 +275,19 @@ def query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResul
...
+            if not self.text_key:
+                node = metadata_dict_to_node(
+                    {"_node_content": hit["entity"].get("_node_content", None)}
+                )
+            else:
+                try:
+                    text = hit["entity"].get(self.text_key)
+                except Exception as e:
+                    print("The passed in text_key value does not exist in the retrieved entity.")
+                    print(e)
```
Comment (logan-markewich) [inline llama_index/vector_stores/milvus.py:287]: Should this just raise a `ValueError()` instead? There's no text to make the node 👀

Comment (ytang07) [reply llama_index/vector_stores/milvus.py]: makes sense, i'm not familiar with the natives yet, accepted

File: llama_index/vector_stores/milvus.py (hunk 1 above)
Comment (ytang07) [reply llama_index/vector_stores/milvus.py]: accepted

**Comment by logan-markewich** [comment]
Thanks!

**Comment by ytang07** [comment]
thank you for helping me with this Logan!

**Comment by mahimairaja** [comment]
Good work @ytang07!

**Pull request merged**